test/data/lightning/test_datamodule.py

```diff
@@ -128,8 +128,7 @@ def test_lightning_dataset(get_dataset, strategy_type):
                                    'pin_memory=True, '
                                    'persistent_workers=False)')
 
-        with pytest.warns(UserWarning, match="defined a `validation_step`"):
-            trainer.fit(model, datamodule)
+        trainer.fit(model, datamodule)
```

Comment:
Since pl raises the warning on rank zero only, let's catch the warning only on rank zero in the test case.

Reply:
Done